modules/server/src/main/java/org/jboss/wsf/stack/cxf/deployment/WSDLFilePublisher.java

```diff
@@ -118,28 +118,53 @@ public void publishWsdlFiles(QName serviceName, String wsdlLocation, Bus bus, Li
 
    public void unpublishWsdlFiles()
    {
-      try
+      String deploymentName = dep.getCanonicalName();
+
+      if (deploymentName.startsWith("http://"))
       {
-         File parentDir = new File(serverConfig.getServerDataDir().getCanonicalPath() + "/wsdl");
-         ArchiveDeployment deployment = dep;
-         while (deployment.getParent() != null)
-         {
-            deployment = deployment.getParent();
-         }
-         String deploymentName = deployment.getCanonicalName();
-         if (deploymentName.startsWith("http://"))
-         {
-            deploymentName = deploymentName.replace("http://", "http-");
-         }
-         File targetDir = new File(parentDir, deploymentName);
-         FileUtils.removeDir(targetDir);
+         deploymentName = deploymentName.replace("http://", "http-");
       }
-      catch (IOException e)
-      {
+
+      try {
+
+         File wsdlRootDir = new File(serverConfig.getServerDataDir().getCanonicalPath()
+             + "/wsdl/");
+         File ancestorDir = new File(serverConfig.getServerDataDir().getCanonicalPath()
+             + "/wsdl/" + deploymentName);
+
+         // create directory list to be deleted
+         List<File> dirList = new ArrayList<File>();
+         String wsdlRootDirStr = wsdlRootDir.getCanonicalPath();
+         do {
+            dirList.add(ancestorDir);
+            ancestorDir = ancestorDir.getParentFile();
+         } while(!wsdlRootDirStr.equals(ancestorDir.getAbsolutePath()));
```

Comment:
Is it right to compare an absolute path here with the canonical path of the initial wsdlRootDir? Btw, not !wsdlRootDir.equals(ancestorDir) ?